### PR TITLE
Add top level filters for filtering by room id.

### PR DESCRIPTION
**Pull request by NegativeMjark**
Based on #245.

### CODE REVIEW

File: api/client-server/definitions/sync_filter.yaml [hunk 1]
```diff
@@ -20,8 +20,27 @@ properties:
...
     description: The presence updates to include.
+  account_data:
+    allOf:
+    - $ref: event_filter.yaml
+    description: The user account data that isn't associated with rooms to include.
   room:
     properties:
+      not_rooms:
+        description: A list of room IDs to exclude. If this list is absent then no rooms
+          are excluded. A matching room will be excluded even if it is listed in the 'rooms'
```
Comment (Kegsay) [inline api/client-server/definitions/sync_filter.yaml]: Backticks on `'rooms'` please.

**Comment by Kegsay** [comment]
We're doing this for the purpose of guest access. We should probably mention that these filters will determine the rooms, or were you planning on doing that in another PR? It feels a little ming to be putting the global filter definition at the same level as the `ephemeral` and `account_data` keys. Perhaps we should use a new keyword like `global`?

File: api/client-server/definitions/sync_filter.yaml (hunk 1 above)
Comment (NegativeMjark) [inline api/client-server/definitions/sync_filter.yaml]: Fixed 9b4a37f

**Comment by NegativeMjark** [comment]
I was going to leave it to another PR to explain how they were used in guest access.

**Comment by Kegsay** [comment]
Okay. Thoughts about shoving this to a `global` section?

**Comment by NegativeMjark** [comment]
I thought using the hierarchical position of the filters to indicate which parts of the response they applied to made it clearer than adding a ``"global"`` section. I would regret that if we ever came to add a conflicting key name.

**Comment by Kegsay** [comment]
Okay, that seems like a good enough reason to keep them where they are. If it turns out it's a mistake, we can always change it :) LGTM

**Pull request merged**